Update Shared Components for Date-Only and Time-Only Tests

labkey-danield opened the pull request:
#### Rationale
Updating shared components used in LKSM Date-only, Time-only and DateTime tests.

The WebDriverWrapper.getCurrentDateFormatString needed to be renamed top getCurrentDateTimeFormatString. Methods getCurrentDateFormatString and getCurrentTimeFormatString were added.

Previously DateTime helpers were named using only the word _date_. With the introduction of date-only field types this would causes confusion. This PR renames the ReactDatePicker to ReactDateTimePicker and is now used as the picker for DateTime and Date-only or Time-only fields. There wasn't much change needed to the component other than to (basically) expose methods that were previously private.

These three pickers are all the ReactDateTimePicker:
![Screenshot 2024-04-16 at 9 00 59 AM](https://github.com/LabKey/testAutomation/assets/12738200/91492361-e062-4252-a59d-da6827796073)
![Screenshot 2024-04-16 at 9 01 09 AM](https://github.com/LabKey/testAutomation/assets/12738200/5815beeb-5dff-4756-990c-ab41e8dac74c)
![Screenshot 2024-04-16 at 9 01 19 AM](https://github.com/LabKey/testAutomation/assets/12738200/415ef30d-7a15-46c1-9e5f-105e52183b97)

DetailTableEdit.setDateTimeField method has been added. This method can be used to set DateTime, Date-only or Time-only fields. The second parameter is an object type and is used to identify the field type. If a string is passed in it is used as a literal value and is simply typed into the text box. The javadoc comments above the method explain this. 

This approach of a single method to be used to set the fields is repeated in various other forms, pages, etc...

EditableGrid.setCellValue now has an overload that takes that indicates if/how the cell update should be validated. This is needed because date and time values have formatting that is applied and can be different from the text sent in. 

#### Related Pull Requests
* https://github.com/LabKey/limsModules/pull/147

#### Changes
* Disambiguate DateTime helpers from Date-only and Time-only helpers.
* Add getters for date and time format settings.
* Exposed selectDate and selectTime methods in ReactDateTimePicker
* Updated ReactDateTimePicker.clear to use the button to clear.
* Updated selectTime method to be smarter about identifying the format of the time in the dropdown list.
* Update EditableGrid to have the option to not wait for an exact match to the value sent in (date and time formatting).

labkey-danield committed:
Rename ReactDatePicker to ReactDateTimePicker.
Update old reference to ReactDatePicker to be ReactDateTimePicker.

labkey-danield committed:
Rename ReactDateInputFinder to ReactDateTimeInputFinder.
Update old reference to ReactDateInputFinder to be ReactDateTimeInputFinder.

labkey-danield committed:
Rename WebDriverWrapper.getCurrentDateFormat to getCurrentDateTimeFormat
Added WebDriverWrapper.getCurrentDateFormat and WebDriverWrapper.getCurrentTimeFormat

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Rename WebDriverWrapper.getCurrentDate,Time,orDateTimeFormat methods to getContainer...
Added test for assay design.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Add date and TIme tests to the SampleType.
Update the editable grid to work with a date-only and time-only field.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Add overload to EditableGrid that gives option of checking for the value entered.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Create and update sample with date and time only fields.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Update ReactDatePicker to have methods to either pick a date or a time or enter a date or time as text.
Update BatchUpdateSamplesDialog to have methods to set a DateTime, Date-only or Time-only field.
Updated EditableGrid to use correct ReactDateTimePicker methods is picking a value.
Added date and time test that bulk updates samples.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Update the clear method for the DateTime picker.
Add test for updating a sample from the detail page.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Reverting name from getContainerDate... to getCurrentDate...

labkey-danield committed:
Simplifying the helper methods that set a DataTime, Date or Time field on various pages & components.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-tchad commented on src/org/labkey/test/components/ui/grids/DetailTableEdit.java:
```diff
@@ -351,6 +355,64 @@ public DetailTableEdit clearSelectValue(String fieldCaption, boolean waitForSele
+        }
+        else if(dateTime instanceof String setValue)
+        {
+            dateTimePicker.set(setValue, true);
+            // When setting DateTime, Date-only or Time-only fields by text there is a slight delay as any formatting is applied.
+            getWrapper().sleep(500);
```
Should this sleep (or some sort of wait) be baked in to `ReactDateTimePicker.set`?

labkey-danield replied:
Removed from here.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Implementing code review feedback.
Changed the generic select method in the ReactDateTimePicker to take an Object parameter and depending on the type will use the appropriate picker.
Added a better check in the ReactDateTimePicker.set method.
Removed common code from AssayImportBasePage, BatchUpdateSamplesDialog.
Changed BaseSettingsPage.reset to wait for the page (removed various sleeps in some tests).
Updated EntityBulkInsertDialog to set DateTime, Date or Time fields.
Updated tests to work around "won't fix" issues.

labkey-danield committed:
Merge branch 'develop' into fb_dateAndTimeTests

labkey-danield committed:
Simplified the wait when using text to set a value in the ReactDateTimePicker.
Updated expected error messages.
Changed inferred field from DateTime to Time.
